Auth0/AuthProvider.swift

```diff
@@ -35,8 +35,27 @@ import UIKit
          ...
          return transaction
      }
+
  ```
  */
 public protocol AuthProvider {
     func login(withConnection connection: String, scope: String, parameters: [String: Any]) -> NativeAuthTransaction
+
+    /**
+     Some IdP authenticators like Auth0Twitter rely upon accessibility of on device social accounts.
+     If these are not accessible/available this pre-login check lets you know you should provide
+     alternative authentication means such as Auth0.WebAuth.
+     
+     - returns: Bool
+     */
+    func canUseNative() -> Bool
+}
+
+/**
+ Extension to provide default(s)
+ */
+public extension AuthProvider {
```

Comment:
Originally I didn't have a default as felt it's important for developers implementing to understand what it did.  Then I thought the most common use case (based on our use case) would be `true` so perhaps this is a convenience.  So i'll go back to original thoughts 👍